src/registry/map/preview.tsx

```diff
@@ -37,10 +39,27 @@ const Preview: React.FC<ComponentPreviewProps<MapComponentSchema>> = ({component
     validate = {},
     defaultZoom,
     initialCenter = {},
+    tileLayerIdentifier,
   } = component;
+  const {getMapTileLayers} = useContext(BuilderContext);
+  const {value: options, loading, error} = useAsync(async () => await getMapTileLayers(), []);
+  if (error) {
+    throw error;
+  }
+
   const {required = false} = validate;
   const {lat = 52.1326332, lng = 5.291266} = initialCenter;
   const zoom = defaultZoom ?? 8;
+
+  const tileLayerUrl = (): string => {
+    if (!tileLayerIdentifier || loading || !options?.length) {
+      return TILE_LAYER_RD.url;
+    }
+    return (
+      options?.find(tileLayer => tileLayer.identifier === tileLayerIdentifier)?.url ??
+      TILE_LAYER_RD.url
+    );
```

Comment:
this should cover `options` being an empty array too, right? And even `tileLayerIdentifier` being `undefined`

Reply:
Those checks are done on line 55. With the `!tileLayerIdentifier` for undefined (or otherwise falsey values), and `!options?.length` for undefined or empty (array size 0). If either one of those is true the function will return the `TILE_LAYER_RD.url` prematurely

Reply:
Ah check, yeah i've changed this a bit, by removing the `?.` from line 58 (because, indeed, at that point options is at least defined as an array).

Yeah your assumption is right. The guard clause isn't really needed..
I do prefer to check the validity of `tileLayerIdentifier` outside the `.find()`. If we don't have to loop over data, we probably shouldn't. I also kinda like a early return, but in this case it might be a bit extreme..

So yeah, we can remove the guard and turn this into a one-liner :)

Reply:
Ive turned it into a one-liner